File: fleet-pay-export/app/src/models/new_company.ts
import { CompanyBusinessHours } from './company_business_hours';
import { NotificationSettings } from './notification_settings';
import { Contact } from './contact';

export interface NewCompany {
  name: string,
  profilePicture?: string | null,
  emails?: string[] | null,
  phoneNumbers?: string[] | null,
  businessHours?: CompanyBusinessHours | null,
  notificationSettings?: NotificationSettings | null,
  accountingDepartmentContact?: Contact | null,
  // --- start extra_fields code ---

  // --- end extra_fields code ---
}

File: 1 fleet-pay-export/flutter/lib/models/new_company.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'company_business_hours.dart';
import 'notification_settings.dart';
import 'contact.dart';
// --- start import code ---
  
// --- end import code ---

part 'new_company.g.dart';
part 'new_company.freezed.dart';

@unfreezed
class NewCompany with _$NewCompany {
  // --- start init code ---
  
  // --- end init code ---

  factory NewCompany({
    required String name,
    String? profilePicture,
    List<String>? emails,
    List<String>? phoneNumbers,
    CompanyBusinessHours? businessHours,
    NotificationSettings? notificationSettings,
    Contact? accountingDepartmentContact,
  }) = _NewCompany;

  factory NewCompany.fromJson(Map<String, dynamic> json) => _$NewCompanyFromJson(json);

  // --- start extension code ---
  
  // --- end extension code ---

}
